src/components/widgets-settings/organizations/organizations.directive.coffee

```diff
@@ -21,9 +34,23 @@ module.controller('SettingOrganizationsCtrl', ($scope, $log, ImpacDashboardsSvc)
     ImpacDashboardsSvc.load().then(
       (config) ->
         $scope.dashboardOrganizations = config.currentDashboard.data_sources
+
         if w.metadata? && w.metadata.organization_ids?
+          count = 0
+          widgetOrgIds = w.metadata.organization_ids
+          currentOrganization = ImpacMainSvc.config.currentOrganization
```

Comment:
This block could do with some refactoring, simplifying and commenting. The code is not very readable, and will cause future confusion for developers.

After looking into for a bit, Im not even sure it's necessary to be iterating over all the `$scope.dashboardOrganizations` and cramming all the logic into all the `if`'s & `else`'s, which Im not even sure we need all of.

I have tried to capture all the cases you have coded for in comments:
```coffeescript
# single mode 
# - has multiple data sources previously selected 
# => deselect all & select current organization
# - none or 1 data source previously selected
# => apply the previously first widget data source organization, or select the first one
```
Is this correct? 

We shouldn't have to "deselect" (e.g set all the key values to false), as this is inside an initialize method, no "checkboxes" are going to be ticked, right?  This saves us having to iterate and set everything to false, and having to use a counter.

The thing with an `ng-model` binding, is that if it is `undefined` or `null`, it's falsey. We should only need to set it when it's been toggled for the first time.

I may not have capture all the cases, but this is a quick proposition of a refactor:

```coffeescript
if w.metadata? && w.metadata.organization_ids?
  if singleOrgMode()
    currentOrganization = ImpacMainSvc.config.currentOrganization
    unless _.isEmpty(currentOrganization)
      w.selectedOrganizations[currentOrganization.uid] = true
    else if (dhbOrg = $scope.dashboardOrganizations[0])?
      w.selectedOrganizations[dhbOrg.uid] = true
```

Anyway I may be completely missing something, but if you could just define & code the logic in a clearer manner that would be awesome! Thanks :)